packages/material-ui-unstyled/src/composeClasses/composeClasses.ts

```diff
@@ -0,0 +1,27 @@
+export default function composeClasses<ClassKey extends string>(
+  slots: Record<ClassKey, Array<string | false>>,
+  getUtilityClass: (slot: string) => string,
```

Comment:
I'm a bit confused about the naming here. We say we call it with a `slot`. But then at the same time we're also calling the keys of `slots` `slot` though we aren't calling `getUtilityClass` with it.

For example, in `composeClasses({ root: ['root', 'disabled'] }, getUtilityClass)`, `composeClasses` is calling `getUtilityClass('disabled')`. But `disabled` isn't a slot as far as I can tell.

Is it more accurate to say that `getUtilityClass` is actually `getUtilityClassName` (since we're passing it to `className`) and name the paramter `utilityClass` instead of `slot`? In summary:
```diff
-getUtlityClass(slot: string)
+getUtlityClassName(utilityClass: string)
```

Reply:
Agree, it can be invoked with slot, but also with prop, state etc. I agree with the proposed changes.

Reply:
Hm, let me do this in another PR since we have other places where we're calling something expecting a "slot" with a "class" e.g. `generateUtilityClasses` calls `generateUtilityClass('root')` (a slot) and `generateUtilityClass('trackInverted')` (a class).

At the same time we're assigning `*UnstyledClasses` to return value from `generateUtilityClasses` which implies that it should be named `*UnstyledUtilityClasses`. And then we're calling `composeClasses` with "utility classes". I'm either missing a bigger picture change that makes uses of the existing abstraction or we've abstracted before we need it.

Reply:
Leaving as unresolved for a future PR.